back/spec/services/local_project_copy_service_spec.rb

```diff
@@ -312,4 +312,243 @@
+              }
+            },
+            lBKtfcN6z2: {
+              type: {
+                resolvedName: 'Image'
+              },
+              nodes: [],
+              props: {
+                alt: '',
+                dataCode: image1.code.to_s
+              },
+              custom: {
+                title: {
+                  id: 'app.containers.admin.ContentBuilder.image',
+                  defaultMessage: 'Image'
+                }
+              },
+              hidden: false,
+              parent: 'vtHcoDw7fg',
+              isCanvas: false,
+              displayName: 'Image',
+              linkedNodes: {}
+            },
+            mjilnZY6iY: {
+              type: {
+                resolvedName: 'TwoColumn'
+              },
+              nodes: [],
+              props: {
+                columnLayout: '1-1'
+              },
+              custom: {
+                title: {
+                  id: 'app.containers.admin.ContentBuilder.twoColumnLayout',
+                  defaultMessage: '2 column'
+                },
+                hasChildren: true
+              },
+              hidden: false,
+              parent: 'ROOT',
+              isCanvas: false,
+              displayName: 'TwoColumn',
+              linkedNodes: {
+                left: 'TMv2pUz2Uw',
+                right: 'AhfqQU7pyq'
+              }
+            },
+            vtHcoDw7fg: {
+              type: {
+                resolvedName: 'Container'
+              },
+              nodes: [
+                'lBKtfcN6z2'
+              ],
+              props: {},
+              custom: {},
+              hidden: false,
+              parent: 'a0g_vKCvY6',
+              isCanvas: true,
+              displayName: 'Container',
+              linkedNodes: {}
+            },
+            xQBSurFNOw: {
+              type: {
+                resolvedName: 'Image'
+              },
+              nodes: [],
+              props: {
+                alt: '',
+                dataCode: image2.code.to_s
+              },
+              custom: {
+                title: {
+                  id: 'app.containers.admin.ContentBuilder.image',
+                  defaultMessage: 'Image'
+                }
+              },
+              hidden: false,
+              parent: 'TMv2pUz2Uw',
+              isCanvas: false,
+              displayName: 'Image',
+              linkedNodes: {}
+            }
+          }
+        }
+
+        layout.update(craftjs_jsonmultiloc: craftjs)
+        copied_project = service.copy(layout.content_buildable)
+
+        new_craftjs = copied_project.content_builder_layouts.first.craftjs_jsonmultiloc
+        new_image_codes = []
+
+        new_craftjs.each_key do |locale|
+          new_craftjs[locale].each_key do |node|
+            next unless new_craftjs[locale][node]['type']['resolvedName'] == 'Image'
+
+            new_image_codes << new_craftjs[locale][node]['props']['dataCode']
+          end
+        end
```

Comment:
This will probably become a method at the model level (or something like this), that returns a collection of all layout images currently associated with the layout.

This is likely to be useful in many places, including [CL-2753 BE Job to clean up layout images not associated with any layout](https://citizenlab.atlassian.net/browse/CL-2753)